{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import paramiko\n",
    "from scp import SCPClient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "client = paramiko.SSHClient()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "client.set_missing_host_key_policy(paramiko.AutoAddPolicy())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "client.connect('ebi-cli',username='jkchang',password = 'David1988')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "sftp_client = client.open_sftp()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "remote_file = sftp_client.open('/net/isilonP/public/rw/homes/tc_cm01/metabolights/prod/studies/stage/private/MTBLS18/i_Investigation.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ONTOLOGY SOURCE REFERENCE\n",
      "\n",
      "Term Source Name\t\"OBI\"\t\"NCBITAXON\"\t\"MS\"\t\"BTO\"\t\"CHMO\"\t\"MS\"\t\"UO\"\t\"EFO\"\t\n",
      "\n",
      "Term Source File\t\"http://data.bioontology.org/ontologies/OBI\"\t\"http://data.bioontology.org/ontologies/NCBITAXON\"\t\"http://data.bioontology.org/ontologies/MS\"\t\"http://data.bioontology.org/ontologies/BTO\"\t\"http://data.bioontology.org/ontologies/CHMO\"\t\"http://data.bioontology.org/ontologies/MS\"\t\"http://data.bioontology.org/ontologies/UO\"\t\"http://data.bioontology.org/ontologies/EFO\"\t\n",
      "\n",
      "Term Source Version\t\"21\"\t\"4\"\t\"86\"\t\"26\"\t\"16\"\t\"129\"\t\"42\"\t\"146\"\t\n",
      "\n",
      "Term Source Description\t\"Ontology for Biomedical Investigations\"\t\"National Center for Biotechnology Information (NCBI) Organismal Classification\"\t\"Mass Spectrometry Ontology\"\t\"BRENDA Tissue and Enzyme Source Ontology\"\t\"Chemical Methods Ontology\"\t\"Mass Spectrometry Ontology\"\t\"Units of Measurement Ontology\"\t\"Experimental Factor Ontology\"\t\n",
      "\n",
      "INVESTIGATION\n",
      "\n",
      "Investigation Identifier\t\"MTBLS18\"\n",
      "\n",
      "Investigation Title\t\"Investigation\"\n",
      "\n",
      "Investigation Description\t\"\"\n",
      "\n",
      "Investigation Submission Date\t\"\"\n",
      "\n",
      "Investigation Public Release Date\t\"\"\n",
      "\n",
      "Comment[Created With Configuration]\t\"\"\n",
      "\n",
      "Comment[Last Opened With Configuration]\t\"\"\n",
      "\n",
      "INVESTIGATION PUBLICATIONS\n",
      "\n",
      "Investigation PubMed ID\t\"\"\n",
      "\n",
      "Investigation Publication DOI\t\"\"\n",
      "\n",
      "Investigation Publication Author List\t\"\"\n",
      "\n",
      "Investigation Publication Title\t\"\"\n",
      "\n",
      "Investigation Publication Status\t\"\"\n",
      "\n",
      "Investigation Publication Status Term Accession Number\t\"\"\n",
      "\n",
      "Investigation Publication Status Term Source REF\t\"\"\n",
      "\n",
      "INVESTIGATION CONTACTS\n",
      "\n",
      "Investigation Person Last Name\t\"\"\n",
      "\n",
      "Investigation Person First Name\t\"\"\n",
      "\n",
      "Investigation Person Mid Initials\t\"\"\n",
      "\n",
      "Investigation Person Email\t\"\"\n",
      "\n",
      "Investigation Person Phone\t\"\"\n",
      "\n",
      "Investigation Person Fax\t\"\"\n",
      "\n",
      "Investigation Person Address\t\"\"\n",
      "\n",
      "Investigation Person Affiliation\t\"\"\n",
      "\n",
      "Investigation Person Roles\t\"\"\n",
      "\n",
      "Investigation Person Roles Term Accession Number\t\"\"\n",
      "\n",
      "Investigation Person Roles Term Source REF\t\"\"\n",
      "\n",
      "STUDY\n",
      "\n",
      "Study Identifier\t\"MTBLS18\"\n",
      "\n",
      "Study Title\t\"Metabolic and transcriptional response of Arabidopsis thaliana wildtype and mutants to Phytophtora infestans\"\n",
      "\n",
      "Study Description\t\"The oomycete P. infestans is the causal agent of late blight, the most devastating potato disease. In contrast to potato, A. thaliana is able to successfully prevent colonization of the pathogen due to a multi-layered nonhost resistance. Several mutants have been isolated which are impaired in penetration resistance. A mutation in the gene PEN2, which encodes for an enzyme involved in indole glucosinolate metabolism, results in the loss of penetration resistance against P. infestans. Despite its ability to penetrate epidermal cells of pen2 mutant plants, P. infestans is still not able to colonize these plants. Additional mutants were isolated which show enhanced defense responses upon infection with P. infestans: pen2erp1 and pen2erp2, and backcrossed mutants erp140 and erp2D. We used 6 different plant lines, the wildtype-like gl1, and the five different mutants (pen2, pen2erp1, pen2erp2, erp2D, erp140). The plants were either infected with P. infestans spores or treated with water as control, and harvested 6 h and 12 h after treatment. The experiment was repeated 3 times with different P. infestans cultures, resulting in biological triplicates, for an overall of 6 x 2 x 2 x 3 = 72 samples.\"\n",
      "\n",
      "Study Submission Date\t\"2015-06-08\"\n",
      "\n",
      "Study Public Release Date\t\"2016-06-02\"\n",
      "\n",
      "Study File Name\t\"s_mtbls18.txt\"\n",
      "\n",
      "Comment[Study Grant Number]\t\"Priority Program 1212\"\n",
      "\n",
      "Comment[Study Funding Agency]\t\"Deutsche Forschungsgemeinschaft\"\n",
      "\n",
      "Comment[Manuscript Licence]\t\"CC BY 3.0\"\n",
      "\n",
      "Comment[Experimental Metadata Licence]\t\"CC0\"\n",
      "\n",
      "Comment[Data Repository]\t\"MetaboLights and ArrayExpress\"\n",
      "\n",
      "Comment[Data Record Accession]\t\"MTBLS18 and E-MTAB-3287\"\n",
      "\n",
      "Comment[Data Record URI]\t\"http://www.ebi.ac.uk/metabolights/MTBLS18 and https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3287/\"\n",
      "\n",
      "Comment[Supplementary Information File Name]\t\"\"\n",
      "\n",
      "Comment[Supplementary Information File Type]\t\"\"\n",
      "\n",
      "Comment[Supplementary File URI]\t\"\"\n",
      "\n",
      "Comment[AEExperimentType]\t\"transcription profiling by array\"\n",
      "\n",
      "STUDY DESIGN DESCRIPTORS\n",
      "\n",
      "Study Design Type\t\"chemically induced mutation\"\t\"Phytophthora infestans 90128\"\t\"time series design\"\t\"disease state design\"\t\"infection\"\t\"ultra-performance liquid chromatography-mass spectrometry\"\t\"diode array detector\"\t\"untargeted metabolites\"\n",
      "\n",
      "Study Design Type Term Accession Number\t\"http://www.ebi.ac.uk/efo/EFO_0000370\"\t\"\"\t\"http://www.ebi.ac.uk/efo/EFO_0001779\"\t\"http://www.ebi.ac.uk/efo/EFO_0001756\"\t\"http://www.ebi.ac.uk/efo/EFO_0000544\"\t\"http://purl.obolibrary.org/obo/CHMO_0000715\"\t\"http://purl.obolibrary.org/obo/CHMO_0002503\"\t\"\"\n",
      "\n",
      "Study Design Type Term Source REF\t\"EFO\"\t\"\"\t\"EFO\"\t\"EFO\"\t\"EFO\"\t\"CHMO\"\t\"CHMO\"\t\"\"\n",
      "\n",
      "STUDY PUBLICATIONS\n",
      "\n",
      "Study PubMed ID\t\"23072470\"\n",
      "\n",
      "Study Publication DOI\t\"10.1111/tpj.12046\"\n",
      "\n",
      "Study Publication Author List\t\"Kopischke M, Westphal L, Schneeberger K, Clark R, Ossowski S, Wewer V, Fuchs R, Landtag J, Hause G, DÃ¶rmann P, Lipka V, Weigel D, Schulze-Lefert P, Scheel D, Rosahl S.\"\n",
      "\n",
      "Study Publication Title\t\"Impaired sterol ester synthesis alters the response of Arabidopsis thaliana to Phytophthora infestans.\"\n",
      "\n",
      "Study Publication Status\t\"Published\"\n",
      "\n",
      "Study Publication Status Term Accession Number\t\"\"\n",
      "\n",
      "Study Publication Status Term Source REF\t\"\"\n",
      "\n",
      "STUDY FACTORS\n",
      "\n",
      "Study Factor Name\t\"Genotype\"\t\"Treatment\"\t\"Sampling time\"\t\"Replicate\"\n",
      "\n",
      "Study Factor Type\t\"genotype\"\t\"plant treatment\"\t\"timepoint\"\t\"biological replicate\"\n",
      "\n",
      "Study Factor Type Term Accession Number\t\"http://www.ebi.ac.uk/efo/EFO_0000513\"\t\"http://purl.obolibrary.org/obo/EO_0001001\"\t\"http://www.ebi.ac.uk/efo/EFO_0000724\"\t\"http://www.ebi.ac.uk/efo/EFO_0002091\"\n",
      "\n",
      "Study Factor Type Term Source REF\t\"EFO\"\t\"EFO\"\t\"EFO\"\t\"EFO\"\n",
      "\n",
      "STUDY ASSAYS\n",
      "\n",
      "Study Assay File Name\t\"a_mtbls18_NEG_mass_spectrometry.txt\"\t\"a_mtbls18_POS_mass_spectrometry.txt\"\t\"a_mtbls18_UV.txt\"\n",
      "\n",
      "Study Assay Measurement Type\t\"metabolite profiling\"\t\"metabolite profiling\"\t\"metabolite profiling\"\n",
      "\n",
      "Study Assay Measurement Type Term Accession Number\t\"http://purl.obolibrary.org/obo/OBI_0000366\"\t\"http://purl.obolibrary.org/obo/OBI_0000366\"\t\"http://purl.obolibrary.org/obo/OBI_0000366\"\n",
      "\n",
      "Study Assay Measurement Type Term Source REF\t\"OBI\"\t\"OBI\"\t\"OBI\"\n",
      "\n",
      "Study Assay Technology Type\t\"mass spectrometry\"\t\"mass spectrometry\"\t\"mass spectrometry\"\n",
      "\n",
      "Study Assay Technology Type Term Accession Number\t\"http://purl.obolibrary.org/obo/OBI_0000470\"\t\"http://purl.obolibrary.org/obo/OBI_0000470\"\t\"http://purl.obolibrary.org/obo/OBI_0000470\"\n",
      "\n",
      "Study Assay Technology Type Term Source REF\t\"OBI\"\t\"OBI\"\t\"OBI\"\n",
      "\n",
      "Study Assay Technology Platform\t\"micrOTOF-Q II ESI-Qq-TOF (Bruker)\"\t\"micrOTOF-Q II ESI-Qq-TOF (Bruker)\"\t\"micrOTOF-Q II ESI-Qq-TOF (Bruker)\"\n",
      "\n",
      "STUDY PROTOCOLS\n",
      "\n",
      "Study Protocol Name\t\"Sample collection\"\t\"Extraction\"\t\"Chromatography\"\t\"Mass spectrometry\"\t\"Data transformation\"\t\"Metabolite identification\"\n",
      "\n",
      "Study Protocol Type\t\"Sample collection\"\t\"Extraction\"\t\"Chromatography\"\t\"Mass spectrometry\"\t\"Data transformation\"\t\"Metabolite identification\"\n",
      "\n",
      "Study Protocol Type Term Accession Number\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Type Term Source REF\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Description\t\"Total RNA was isolated according to[1] and purified using the RNeasy Plant Mini Kit. Labelling was performed by AROS APPLIED BIOTECHNOLOGY (Aarhus, Denmark). Nucleic acid hybridisation was performed by AROS APPLIED BIOTECHNOLOGY (Aarhus, Denmark).\n",
      "\n",
      "</p>\n",
      "\n",
      "Blumeria graminis f. sp. hordei (Bgh) was propagated on barley (cultivar Ingrid, line I-10). Arabidopsis plants were inoculated using a settling tower, as previously described[2]. To score pathogen invasion and cell death in the epidermis, 3 leaf samples of each genotype were taken at 2 dpi and analyzed microscopically under UV light[3]. 100 interaction sites were scored per leaf. Microscopic analyses were carried out at 6 dpi, and macroscopic phenotypes were monitored at 9 dpi.\n",
      "\n",
      "</p>\n",
      "\n",
      "The growth of G. cichoracearum (strain UCSC1) and infection of A. thaliana was performed as described by Adam et al. (1999). Microscopic analyses were carried out at 7 dpi. A minimum of 9 leaves of different plants from each genotype was monitored in each experiment.\n",
      "\n",
      "</p>\n",
      "\n",
      "Bacteria were infiltrated into Arabidopsis rosette leaves using a needleless syringe. For the determination of bacterial growth, P. syringae pv. tomato (Pto DC3000 and Pto DC3000 avrRPM1) was infiltrated at 1 x 105 cfu/ml and P. syringae pv. phaseolicola (Pph) was infiltrated at 1 x 108 cfu/ml.  Immediately after infiltration (0 dpi) and at 4 dpi, bacterial titers were determined. In a minimum of 3 experiments, 3 leaf disks of 9 mm in diameter of 3 different plants per genotype were sampled and ground in 0.2 ml of 10 mm MgCl2. Bacterial growth was determined by plating serial dilutions (3 technical replicates) on nutrient agar plates and counting the cfu after incubation for 2 days at 28 Â°C.\n",
      "\n",
      "</p>\n",
      "\n",
      "For callose assays, Pto DC3000 avrRPM1 was infiltrated at 5 x 105 cfu/ml and aniline blue staining was performed 2 dpi.\n",
      "\n",
      "</p>\n",
      "\n",
      "Ref:</br>\n",
      "\n",
      "[1] Ahn Ji H. RNA Extraction from Arabidopsis for Northern Blots and Reverse Transcriptase-PCR. Adapted from Arabidopsis: A Laboratory Manual (eds. Weigel and Glazebrook). CSHL Press, Cold Spring Harbor, NY, USA, 2002. http://dx.doi.org/10.1101/pdb.prot5295</br>\n",
      "\n",
      "[2] Zimmerli L, Stein M, Lipka V, Schulze-Lefert P, Somerville S. Host and non-host pathogens elicit different jasmonate/ethylene responses in Arabidopsis. Plant J. 2004 Dec;40(5):633-46. doi:10.1111/j.1365-313X.2004.02236.x. PMID:15546348</br>\n",
      "\n",
      "[3] Lipka V, Dittgen J, Bednarek P, Bhat R, Wiermer M, Stein M, Landtag J, Brandt W, Rosahl S, Scheel D, Llorente F, Molina A, Parker J, Somerville S, Schulze-Lefert P. Pre- and postinvasion defenses both contribute to nonhost resistance in Arabidopsis. Science. 2005 Nov 18;310(5751):1180-3. doi:10.1126/science.1119409. PMID:16293760</br>\"\t\"Plant material was homogenized in liquid nitrogen using a pestle and mortar and aliquots of 50 +/- 5 mg were weighed into pre-cooled 2-ml, round bottom tubes. After addition of 100 Âµl methanol/water, 80/20 (v/v) pre-cooled at â40 Â°C the samples were immediately vortexed for 15 s, sonicated for 15 min at 20 Â°C and centrifuged for 10 min at 19000 x g. The supernatants were transferred to new 2-ml tubes and the remaining pellets subjected to a second extraction using 100 Âµl methanol/water, 80/20 (v/v). The combined extracts were filtered using 0.2-Âµm PTFE syringe filters. 3 extracts (technical replicates) were prepared for each of the 72 leaf pools.\"\t\"Chromatographic separations were performed on an Acquity UPLC system (Waters) equipped with a HSS T3 column (100 x 1.0 mm, particle size 1.8 Âµm; Waters) with a flow rate of 150 Âµl/min at 40 Â°C column temperature using the following gradient program: 0-60 s, isocratic 95% A (water/formic acid, 99.9/0.1 (v/v)), 5% B (acetonitrile/formic acid, 99.9/0.1 (v/v)); 60-360 s, linear from 5% to 30% B; 360-600 s, linear from 30% to 95% B; 600-720 s isocratic 95% B. The injection volume was 2.0 Âµl (full loop injection).\n",
      "\n",
      "</p>\n",
      "\n",
      "Elution of metabolites was detected with the help of a Waters Acquity UPLC Photodiode Array (PDA) detector. The DAD signal was acquired from 190 to 500 nm (step 0.25 nm) with a resolution of 0.25 nm, at a frequency of 20 points/s.\"\t\"Eluting compounds were detected with a spectra rate of 3 Hz from m/z 50â1000 using a MicrOTOF-Q-II hybrid quadrupole time-of-flight mass spectrometer (Bruker, Daltonics) equipped with an Apollo II electrospray ion source in either positive or negative ion mode with the following instrument settings: nebulizer gas, nitrogen, 1.2 bar; dry gas, nitrogen, 8 l/min, 190 Â°C; capillary, 4500 V; end plate offset, -500 V; funnel 1 RF, 200 Vpp; funnel 2 RF, 200 Vpp; hexapole RF, 100 V. Mass spectra were acquired in centroid mode. Mass calibration of individual raw data files was performed on lithium formate cluster ions obtained by automatic infusion of 20 Âµl 10 mM lithium hydroxide in isopropanol/water/formic acid, 49.9/49.9/0.2 (v/v/v) at a gradient time of 720 s using a diverter valve.\"\t\"Raw data files were converted to mzData format using the vendor-specific CompassXport (http://www.bdal.de/) and processed using the XCMS package (http://bioconductor.org/packages/release/bioc/html/xcms.html). XCMS settings for processing LC/MS data with findPeaks.centWave() were prefilter=(3,200); snthr=5; ppm=25; peakwidth=(5,12). For alignment group.density() function with parameters minfrac=0.75 and bw=2 was used. The processing of the CEL files was performed with the R-package simpleaffy.\"\t\"Currently no metabolite identification information included in this study.\"\n",
      "\n",
      "Study Protocol URI\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Version\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Parameters Name\t\"\"\t\"Post Extraction;Derivatization\"\t\"Chromatography Instrument;Column type;Column model;Detector;Signal range;Resolution\"\t\"Scan polarity;Scan m/z range;Instrument;Mass analyzer;Ion source\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Parameters Name Term Accession Number\t\"\"\t\";\"\t\";;;;;\"\t\";;;;\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Parameters Name Term Source REF\t\"\"\t\";\"\t\";;;;;\"\t\";;;;\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Components Name\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Components Type\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Components Type Term Accession Number\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Protocol Components Type Term Source REF\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "STUDY CONTACTS\n",
      "\n",
      "Study Person Last Name\t\"Neumann\"\t\"Rosahl\"\t\"LÃ¼bken\"\t\"Westphal\"\t\"Czamperla\"\t\"Kopischke\"\t\"Scheel\"\n",
      "\n",
      "Study Person First Name\t\"Steffen\"\t\"Sabine\"\t\"Tilo\"\t\"Lore\"\t\"Kathrin\"\t\"Michaela\"\t\"Dierk\"\n",
      "\n",
      "Study Person Mid Initials\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Email\t\"sneumann@ipb-halle.de\"\t\"srosahl@ipb-halle.de\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Phone\t\"+49 345 5582 1409\"\t\"+49 345 5582 1440\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Fax\t\"+49 345 5582 1409\"\t\"+49 345 5582 1409\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Address\t\"Weinberg 3, 06120 Halle, Germany\"\t\"Weinberg 3, 06120 Halle, Germany\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Affiliation\t\"IPB Halle\"\t\"IPB Halle\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Roles\t\"submitter\"\t\"investigator\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Roles Term Accession Number\t\"http://www.ebi.ac.uk/efo/EFO_0001741\"\t\"http://www.ebi.ac.uk/efo/EFO_0001739\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n",
      "Study Person Roles Term Source REF\t\"EFO\"\t\"EFO\"\t\"\"\t\"\"\t\"\"\t\"\"\t\"\"\n",
      "\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    for line in remote_file:\n",
    "        print(line)\n",
    "except:\n",
    "    print('remote_file failed')\n",
    "    remote_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
